### ISSUE TITLE
Diagnostic CodeFix for CS0433: The type TypeName1 exists in both Assembly1 and Assembly2

### ISSUE THREAD

**Issue by MaStr11** [dotnet/roslyn#24099]
**Version Used**: 

VS 15.5.2

**Steps to Reproduce**:

If a projects references two libraries with the same type defined in the same namespace CS0433 is triggered if this type is used. A code fix should be offered that

1. adds an alias to the library reference,
2. adds an corresponding extern alias directive to the document
3. fully qualifies the type (additionally a type alias could be offered as in #24022)

The fix should also be offered for warning `CS0436 The type 'type' in 'assembly' conflicts with the imported type 'type2' in 'assembly'. Using the type defined in 'assembly'.`

**Before the fix**:

Assembly1 File1.cs  
```C#
namespace N1
{
    public class Ambiguous
    {
    }
}
```
Assembly2 File2.cs
```C#
namespace N1
{
    public class Ambiguous
    {
    }
}
```
TestAssembly
References Assembly1 and Assembly2
File3.cs in TestAssembly
```C#
namespace N2
{
    public class Test
    {
        void M()
        {
              new N1.Ambiguous(); //CS0433
        }
    }
}
```

**After the fix**:

TestAssembly
References Assembly1 **reference is aliased with *Assembly1Alias*** and Assembly2
File3.cs in TestAssembly
```C#
extern alias Assembly1Alias;
namespace N2
{
    public class Test
    {
        void M()
        {
              new Assembly1Alias::N1.Ambiguous(); 
        }
    }
}
```
Relates to #23326.

**Comment by CyrusNajmabadi** [comment]
Closing out for lack of customer feedback on this issue.
